File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/entity/Wishlist.java
package com.amakakeru.mangaworld.entity;

import com.amakakeru.mangaworld.entity.IdClasses.WishlistId;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@IdClass(WishlistId.class)
@Table(name = "wishlist")
public class Wishlist implements Serializable {

    @Id
    @Column(name = "user_id")
    private Long userId;

    @Id
    @Column(name = "m_id")
    private Long mId;

    @Column(name = "added_date", nullable = false)
    private java.sql.Timestamp addedDate;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}

File: Frontend/manga-world/src/Model/WishList.ts
import Manga from "./Manga";
import User from "./user";

class wishList {
    userId: number;
    mId: number;
    addedDate: Date;
    user: User;
    manga: Manga;

    constructor(userId: number, mId: number, addedDate: Date, user: User, manga: Manga) { 
        this.userId = userId;
        this.mId = mId;
        this.addedDate = addedDate;
        this.user = user;
        this.manga = manga;
    }
}

export default wishList; 